modules/ROOT/pages/configure-horizontal-autoscaling.adoc

```diff
@@ -0,0 +1,200 @@
+- --balancing-ignore-label=vpc.amazonaws.com/eniConfig
+----
+
+=== Configure Overprovisioning Pod
+
+Configure at least one overprovisioning pod per availability zone. Each overprovisioning pod holds a resource that can accommodate three Mule app pods, which improves the performance when new nodes are needed.
+
+
+
+[NOTE]
+Refer to your managed vendor documentation for cluster autoscaling best practices and over-provisioning configuration details.
+
+
+=== Configure Topology Spread Constraints
+
+Your Mule app deployments which are enabled with autoscaling, are required to customize the kubernetes resources for the topology spread constraints in pod spec by updating the *whenUnsatisfiable* field with `DoNotSchedule`.
+This configuration ensures that autoscaling replicas are evenly spread across zones for high availability.
+
+----
+topologySpreadConstraints:
+  - maxSkew: 1
+    topologyKey: topology.kubernetes.io/zone
+    whenUnsatisfiable: DoNotSchedule
+    labelSelector:
+      matchLabels:
+        rtf.mulesoft.com/id: {{ .Values.id }}
+----
+
+For further details on the configuration, refer to xref:customize-kubernetes-crd.adoc[] documentation.
+
+[WARNING]
+Customizing topology spread with `whenUnsatisfiable: DoNotSchedule`, without the recommended configurations for node groups per availability zone, cluster autoscaler, and overprovisionining, can lead to K8s failing to schedule the replicas with `Pending state` error.
+
+=== Understand Autoscaling Policy
+
+MuleSoft owns and applies the https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/[autoscaling^] policy for your Mule application deployments.
+
+The CPU based HPA policy used for all Mule apps deployed on Runtime Fabric instances, is as follows:
+
+----
+apiVersion: autoscaling/v2
+kind: HorizontalPodAutoscaler
+metadata:
+  name: my-app
+  namespace: app-namespace
+spec:
+  behavior:
+    scaleDown:
+      policies:
+      - periodSeconds: 15
+        type: Percent
+        value: 100
+      selectPolicy: Max
+      stabilizationWindowSeconds: 300
+    scaleUp:
+      policies:
+      - periodSeconds: 180
+        type: Percent
+        value: 100
+      selectPolicy: Max
+      stabilizationWindowSeconds: 0
+  maxReplicas: 3
+  metrics:
+  - resource:
+      name: cpu
+      target:
+        averageUtilization: 70
+        type: Utilization
+    type: Resource
+  minReplicas: 1
+  scaleTargetRef:
+    apiVersion: apps/v1
+    kind: Deployment
+    name: my-app
+
+----
+
+Some points to consider:
+
+* Up to 100% running pods may be added until the maximum configured replicas is reached.
+* The number of pods added is based on the aggregated calculations over the past 180 seconds.
+* The number of pods removed is based on the aggregated calculations over the past 300 seconds.
+* Max scale up profile is 1 -> 2 -> 4 -> 8, where MuleSoft hits 8 replicas in approximately 6 minutes.
+
+The following are few examples of Mule app types that fit CPU based HPA:
```

Comment:
If we have only 2 examples out of which one is not really an app type, let us rephrase this so the 'list' does not look so light.
Something to the order of:
Mule Application types that fit with CPU based HPA are applications that scale based on CPU usage
I would also move this as the first sentence in the "Performance Considerations and Limitations section" and not in the HPA policy section

Mule Applications that scale based on CPU usage would be a good fit with CPU based HPA. For e.g. HTTP/HTTPS apps with async requests, 'B', 'C'. (as informed by Q4 testing).

(We can also add anti-patterns if there is input from Q4)

Reply:
@rithishapadmanabh what does 'B', 'C stand for here?